solr/core/src/java/org/apache/solr/schema/BinaryField.java

```diff
@@ -94,6 +99,10 @@ public IndexableField createField(SchemaField field, Object val) {
       log.trace("Ignoring unstored binary field: {}", field);
       return null;
     }
+    return new org.apache.lucene.document.StoredField(field.getName(), getBytesRef(val));
+  }
+
+  private static BytesRef getBytesRef(Object val) {
```

Comment:
Moved the code for converting `Object val` to `BytesRef` for code reuse (and readability)